src/Rules/Properties/ReadOnlyByPhpDocPropertyAssignRule.php

```diff
@@ -67,7 +72,10 @@ public function processNode(Node $node, Scope $scope): array
 				throw new ShouldNotHappenException();
 			}
 
-			if (strtolower($scopeMethod->getName()) === '__construct' || strtolower($scopeMethod->getName()) === '__unserialize') {
+			if (
+				in_array($scopeMethod->getName(), $this->constructorsHelper->getConstructors($scopeClassReflection), true)
+				|| strtolower($scopeMethod->getName()) === '__unserialize'
+			) {
```

Comment:
seems to be working fine but I saw that the other rules use `Node::getUninitializedProperties` instead. Maybe I should look into that more, but not sure if that means that this 2 rules here have to change more or not..

Reply:
wait, no, this is another type of node, I can't use that.